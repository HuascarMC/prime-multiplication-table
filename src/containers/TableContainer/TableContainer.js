import React from 'react';
import TableRowComponent from '../../components/TableRowComponent/TableRowComponent.js'
var Primable = require('../../tools/primable/Primable.js');

class TableContainer extends React.Component {
 constructor(props) {
  super(props);

  this.state = ({
   primes: [],
   primable: new Primable,
   uniqueKey: 0,
  })
}

generateNewKey() {
 this.state.uniqueKey += 1;
}


getPrimes(number) {
 this.state.primes = this.state.primable.getPrimes(number);
}

 render() {
  this.getPrimes(this.props.number);
   this.generateNewKey();
  return(
  <table className="primes-table" key={this.state.uniqueKey}>
   <TableRowComponent primes={this.state.primes}/>
  </table>
 )
 }
}

export default TableContainer;
