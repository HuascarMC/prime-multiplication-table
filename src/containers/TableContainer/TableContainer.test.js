import React from 'react';
import TableContainer from './TableContainer'
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

it('should render without crashing', () => {
 shallow(<TableContainer />);
})

it('has a primes array state', () => {
 const wrapper = shallow(<TableContainer />);
 const state = wrapper.state();
 expect(state.primes.length).toEqual(0)
})

it('should render table tags', () => {
 const wrapper = shallow(<TableContainer />);
 const table = wrapper.find('.primes-table');
 expect(table.exists()).toEqual(true);
})

it('should match the snapshot', () => {
 const component = renderer.create(
  <TableContainer />
 );
 let tree = component.toJSON();
 expect(tree).toMatchSnapshot();
})


it('should be able to use primable class', () => {
 const wrapper = shallow(<TableContainer />);
 const primable = wrapper.state().primable

 expect(primable).toBeTruthy();
})

it('should be able to use primable getPrimes method', () => {
 const wrapper = shallow(<TableContainer />);
 const primable = wrapper.state().primable

 const result = primable.getPrimes(10)
 expect(result.length).toEqual(5);
})

it('should be able to use primable increaseArray method', () => {
 const wrapper = shallow(<TableContainer />);
 const primable = wrapper.state().primable

 const primesArray = primable.getPrimes(10);
 const increasedArray = primable.increaseArray(primesArray, 3);
 const result = [6, 9, 15, 21, 27];
 expect(increasedArray).toEqual(result);
})

it('should be able to use primable is Prime method', () => {
 const wrapper = shallow(<TableContainer />);
 const primable = wrapper.state().primable

 const result = primable.isPrime(7);
 expect(result).toEqual(true);
})
