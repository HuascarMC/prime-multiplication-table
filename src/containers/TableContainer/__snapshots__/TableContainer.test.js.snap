// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<table
  className="primes-table"
>
  <tbody>
    <tr />
  </tbody>
</table>
`;
